### DISCUSSION TITLE
XMake generator

### DISCUSSION THREAD

**Discussion by ghost** [premake/premake-core#2305] [Ideas]
XMake is predominated in Chinese developers' projects. It's either CMake or XMake, or an optional CMakeLists.txt generated by XMake on their projects. It's very useful if Premake could generate XMake.

**Comment by KyrietS** [comment]
If you want to configure your project in xmake, then why use Premake in the first place? You want to write Lua scripts for Premake so that it can generate different Lua scripts for xmake? 😅 Or am I missing something here?

**Reply by ghost** [reply]
There is an existing project that uses Premake. I want to present it to Chinese developers, so I think if Premake could generate XMake then it will be easier to attract them.

**Reply by KyrietS** [reply]
Alright, unfortunately as for now you would have to prepare the XMake configuration yourself or create such add-on generator from scratch, sorry 🙁

**Comment by Jarod42** [comment]
There is [premake-cmake](https://github.com/Jarod42/premake-cmake) to generate CMakeLists.txt (for CMake).
I don't know xmake and I didn't see any premake-xmake...

**Comment by samsinsane** [comment]
I'm going to close this discussion as this isn't something Premake should do natively. As has already been mentioned, the module system is there if someone wants to make a premake-xmake module, similar to the premake-cmake module.
